src/Main.elm

```diff
@@ -1,10 +1,197 @@
 module Main exposing (main)
 
-import Data.Audience
-import Data.AudienceFolder
-import Html exposing (Html)
+import Browser
+import Data.Audience as Aud
+import Data.AudienceFolder as AudFolders
+import Debug exposing (toString)
+import Html exposing (..)
+import Html.Attributes exposing (..)
+import Html.Events exposing (onClick)
 
 
-main : Html msg
+main : Program () Model Msg
 main =
-    Html.text "There will be app soon!"
+    Browser.sandbox
+        { init = init
+        , update = update
+        , view = view
+        }
+
+
+type alias Model =
+    { folders : List AudFolders.AudienceFolder
+    , opened : Opened
+    , audience : List Aud.Audience
+    , breadcrumbs : Breadcrumbs
+    }
+
+
+type alias Opened =
+    { currentId : Maybe Int
+    , parentId : Maybe Int
+    , usedIdList : List Int
+    , parentName : List String
+    , state : Bool
+    }
+
+
+type alias Breadcrumbs =
+    { breadCrumbName : List String
+    , breadCrumbId : List Int
+    }
```

Comment:
This can go easily out of sync. You can have list of names not matching list of Ids. 
Better type would be `List { breadcrumbName: String, breadcrumbId: Int }`.